pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -70,21 +78,37 @@ def get_profiled_train_dataloader(self, train_dataloader) -> Iterable:
     def prepare_data(self) -> None:
         # on multi-gpu jobs we only want to manipulate (download, etc) on node_rank=0, local_rank=0
         # or in the case where each node needs to do its own manipulation in which case just local_rank=0
-        if self.can_prepare_data():
-            if self.trainer.datamodule is not None:
-                self.trainer.datamodule.prepare_data()
+        should_call_once_per_node = self.trainer.local_rank == 0
+        should_call_once_in_total = self.trainer.local_rank == 0 and self.trainer.node_rank == 0
```

Comment:
these variable names are a little misleading. the local rank and node rank don't determine whether they should be called once per node or once in total. 

the local rank is in the range [0, # of processes) on a particular node. 
the node rank is in the range [0, # of nodes) 

should call once per node or once in total is slightly separate. we could call them local_rank_zero and global_zero